firmware/src/receiver_counters.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * receiver_counters.c
+ * Copyright (C) 2015 Simon Newton
+ */
+
+#include "receiver_counters.h"
+
+static const uint16_t UNINITIALIZED_COUNTER = 0xffff;
```

Comment:
I re-read the MISRA rules:

If a constant is of an unsigned type, it is helpful to avoid ambiguity by applying a “U” suffix. When applied to larger values, the suffix may be redundant (in the sense that it does not influence the type of the constant); however its presence is a valuable contribution towards clarity

So I've added it back